csharp/src/Ice/OutgoingResponseFrame.cs

```diff
@@ -190,19 +186,15 @@ public static OutgoingResponseFrame WithVoidReturnValue(Current current)
                         }
                         else
                         {
-                            // TODO: this code is currently unreachable because the application never gets an incoming
-                            // response frame carrying a system exception - this system exception is always thrown.
-                            Debug.Assert(false);
-
                             OutputStream.Position tail =
                                 OutputStream.WriteEncapsulationHeader(Data,
                                                                       EncapsulationStart,
                                                                       Ice2Definitions.Encoding,
-                                                                      response.Payload.Count,
+                                                                      response.Payload.Count + 2,
```

Comment:
So here we are marshaling an ice1 system exception in an ice2/Encoding "failure" response, where Encoding = response.Encoding (i.e. 1.1).
The size of this encapsulation should be: 2 (encoding) + response.Payload.Count. And response.Payload.Count includes the reply status byte. (All good).

Reply:
Right